crates/precompile/src/blake2.rs

```diff
@@ -17,19 +17,19 @@ pub fn run(input: &Bytes, gas_limit: u64) -> PrecompileResult {
         return Err(Error::Blake2WrongLength.into());
     }
```

Comment:
technically this check also would go below gas check, and then gas check would have to handle empty input...